crates/typst/src/diag.rs

```diff
@@ -228,6 +239,49 @@ impl From<SyntaxError> for SourceDiagnostic {
     }
 }
 
+/// The identifier of a [`SourceDiagnostic`].
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum Identifier {
+    /// Identifier for a built-in compiler warning.
+    Warn(CompilerWarning),
```

Comment:
The `Warn` variants is strangely named since both are warnings. Perhaps it should rather be `Compiler`.

Reply:
This was mostly because there used to be an `Error` variant, but I removed it since I deemed it out of scope for the PR; we can keep it open to adding error identifiers in the future.

Though I'd still accept a change in the name, however just `Compiler` would become ambiguous in that context. Maybe `CompilerWarn` or similar.

Reply:
Resolving due to https://github.com/typst/typst/pull/4459#discussion_r1688231225